### DISCUSSION TITLE
Rain Template <p:divider / p:seperator in pr:menu

### DISCUSSION THREAD

**Discussion by rider87** [primefaces/community#220] [PrimeFaces Templates] [answered]
Hi, 

How can i add <p:divider / p:seperator in pr:menu ?
To create some sections, like below picture:
![image](https://github.com/primefaces/community/assets/9787533/d539a337-1732-4f54-aa87-5badcaa9614b)

Thanks

**Comment by rider87** [comment]
Any ideas?

**Comment by rider87** [comment]
Any feedback here? 

Sorry to say, but I wish we will have still the old forum back. The support is very bad for those issue which I raised here in the new Github forum.

**Comment by kwee27** [comment]
Hey ... I'm pretty new to this, but I had a look through the code and looks like the menu's separator rendering is controlled by the `org.primefaces.rain.component.RainMenuRenderer#encodeSeparator` method. Here's its implementation:
```
    @Override
    protected void encodeSeparator(FacesContext context, Separator separator) throws IOException {
        ResponseWriter writer = context.getResponseWriter();
        String style = separator.getStyle();
        String styleClass = separator.getStyleClass();
        styleClass = styleClass == null ? "Separator" : "Separator " + styleClass;

        //title
        writer.startElement("li", null);
        writer.writeAttribute("class", styleClass, null);
        if(style != null) {
            writer.writeAttribute("style", style, null);
        }

        writer.endElement("li");
    }
```

I have no idea why its been coded like that, but I guess you have 2 choices:

1. when adding your separator, add the `styleClass` attributes directly like this:
`<separator styleClass="ui-divider ui-widget ui-divider-horizontal ui-divider-solid"/>`
There are a whole lot of other classes which may make sense and are applied in the overridden class's implementation. Have a look at that code and see what makes sense to you.

2.  comment out the `org.primefaces.rain.component.RainMenuRenderer#encodeSeparator` method so that the super class's method seems to work.


Note that the menus in the UI Kit on the `http://localhost:8080/rain-5.0.0/menu.xhtml` page aren't rendered through this class and are not affected. So, in those menu's adding `<p:separator />` or `<p:divider />` seems to work fine.

Anyway, hope this is at least a little useful.

**Comment by rider87** [comment]
thank you so much, exactly what I´m looking for.

**Reply by kwee27** [reply, marked as answer]
@rider87 - would you consider this topic `answered`
